src/System.CommandLine/Help/HelpBuilder.cs

```diff
@@ -280,14 +280,44 @@ protected virtual IReadOnlyCollection<string> SplitText(string text, int maxLeng
         /// <summary>
         /// Formats the help rows for a given argument
         /// </summary>
-        /// <param name="commandDef"></param>
+        /// <param name="symbol"></param>
         /// <returns>A new <see cref="HelpItem"/></returns>
-        protected virtual HelpItem ArgumentFormatter(ISymbol commandDef)
+        protected virtual HelpItem ArgumentFormatter(ISymbol symbol)
         {
-            return new HelpItem {
-                Invocation = $"<{commandDef?.Argument?.Name}>",
-                Description = commandDef?.Argument?.Description ?? "",
-            };
+            var argumentName = ArgumentDescriptor(symbol.Argument);
+
+            return new HelpItem
+                   {
+                       Invocation = $"<{argumentName}>",
+                       Description = symbol.Argument?.Description ?? ""
+                   };
+        }
+
+        protected virtual string ArgumentDescriptor(IArgument argument)
+        {
+            var argumentType = argument.Type;
+
+            if (argumentType != null)
+            {
+                if (argumentType.IsConstructedGenericType && 
+                    argumentType.GetGenericTypeDefinition() == typeof(Nullable<>))
+                {
+                    argumentType = argumentType.GetGenericArguments().Single();
+                }
+
+                if (argumentType == typeof(bool))
+                {
+                    return "true|false";
+                }
+
```

Comment:
is there a plan on constructing arg name from Suggestion sources as well? If yes, Is it gonna be in this PR?

Reply:
Yes.